frontend/src/components/NewRunParametersV2.tsx

```diff
@@ -62,6 +62,18 @@ function NewRunParametersV2(props: NewRunParametersProps) {
   const [customPipelineRoot, setCustomPipelineRoot] = useState(props.pipelineRoot);
 
   const [updatedParameters, setUpdatedParameters] = useState({});
+  const inputConverter = (paramStr: string) => {
+    if (paramStr === 'True') {
+      return true;
+    } else if (paramStr === 'False') {
+      return false;
+    } else if (Number(paramStr)) {
+      return Number(paramStr);
+    } else {
+      return paramStr;
+    }
```

Comment:
We shouldn't convert the type purely based on whether it it can be converted to one type or not. For example: If a pipeline template asks for String type, then user provides a value `True`. This `True` should not be boolean type and should be string type.